src/main/java/DungeonController.java

```diff
@@ -0,0 +1,242 @@
+        Random rand = new Random();
+        int startHeight;
+        int startWidth;
+
+        for (int r = 1; r <= rooms; r++) { //1 - top left, 2 - top right, 3 - bottom left, 4 - bottom right
+
+            switch (r) {
+                case 1:
+                    startHeight = rand.nextInt((dungeon.getHeight() / 2 - 6) - 2) + 2;
+                    startWidth = rand.nextInt((dungeon.getWidth() / 2 - 6) - 2) + 2;
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                case 2:
+                    startHeight = rand.nextInt((dungeon.getHeight() / 2 - 6) - 2) + 2;
+                    startWidth = rand.nextInt((dungeon.getWidth() - 7) - (dungeon.getWidth() / 2 + 2)) + (dungeon.getWidth() / 2 + 2);
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                case 3:
+                    startHeight = rand.nextInt((dungeon.getHeight() - 7) - (dungeon.getHeight() / 2 + 2)) + (dungeon.getHeight() / 2 + 2);
+                    startWidth = rand.nextInt((dungeon.getWidth() / 2 - 6) - 2) + 2;
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                case 4:
+                    startHeight = rand.nextInt((dungeon.getHeight() - 7) - (dungeon.getHeight() / 2 + 2)) + (dungeon.getHeight() / 2 + 2);
+                    startWidth = rand.nextInt((dungeon.getWidth() - 7) - (dungeon.getWidth() / 2 + 2)) + (dungeon.getWidth() / 2 + 2);
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                default:
+                    startHeight = rand.nextInt(1);
+                    startWidth = rand.nextInt(1);
+            }
+
+
+            for (int i = 0; i < 5; i++) {
+                //draw room (y, x)
+                dungeonMatrix[i + startHeight][startWidth] = "*"; //left
+                dungeonMatrix[startHeight][i + startWidth] = "*"; //top
+                dungeonMatrix[i + startHeight][startWidth + 4] = "*"; //right
+                dungeonMatrix[startHeight + 4][i + startWidth] = "*"; //bottom
+
+                gridBounds[i + startHeight][startWidth] = 1; //left
+                gridBounds[startHeight][i + startWidth] = 1; //top
+                gridBounds[i + startHeight][startWidth + 4] = 1; //right
+                gridBounds[startHeight + 4][i + startWidth] = 1; //bottom
+
+            }
+            removeDoor(startHeight, startWidth);
+        }
+    }
+
+    private void addGold(int room, int sh, int sw) {
+        golds[room - 1] = new Gold();
+        golds[room - 1].setGoldX(sw + 2);
+        golds[room - 1].setGoldY(sh + 2);
+
+        dungeonMatrix[sh + 2][sw + 2] = golds[room - 1].getGoldSymbol();
+        gridBounds[sh + 2][sw + 2] = 2;
+    }
+
+    private void removeDoor(int sh, int sw) {
+        Random rand = new Random();
+        int wall = rand.nextInt(4);
+
+        switch (wall) {
```

Comment:
Used this case structure twice now, consider enum rather than comments of what they are